### PR TITLE
Validate duplicate bindings across all binding types

### LINKED ISSUE #2893: 🐛 BUG: same name for the queue and its binding generates internal server error (10002)
### Which Cloudflare product(s) does this pertain to?

Queues

### What version of `Wrangler` are you using?

2.12.2

### What operating system are you using?

Mac

### Describe the Bug

```
[[queues.producers]]
  queue = "data"
  binding = "data"
```

produces
```
✘ [ERROR] A request to the Cloudflare API (/accounts/xxxxxxx/workers/scripts/zzzzzz) failed.

  workers.api.error.internal_server [code: 10002]
```

renaming binding to anything else fixes it.

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Fixes #2893

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Tests not necessary because: behaviour already tested, this PR just adds it to more bindings (and adds a type level to check to make sure this doesn't regress)
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: simple bugfix

### CODE REVIEW

File: packages/wrangler/src/config/validation.ts [hunk 1]
```diff
@@ -2597,37 +2599,27 @@ const validateHyperdriveBinding: ValidatorFn = (diagnostics, field, value) => {
...
+			binding,
+			getBindingNames(
+				bindingType === "queues"
+					? config[bindingType]?.producers
+					: config[bindingType]
+			),
+		])
+	);
+
+	console.log(bindingsGroupedByType);
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/config/validation.ts:2622]: is this log needed here?

**Comment by CarmenPopoviciu** [comment]
this LGTM. Left a small comment + don't forget about adding a changeset. Otherwise, good to go

File: packages/wrangler/src/config/index.ts [hunk 2]
```diff
@@ -507,7 +535,7 @@ export function printBindings(bindings: CfWorkerInit["bindings"]) {
 
 	if (unsafe?.metadata !== undefined) {
 		output.push({
-			type: "Unsafe Metadata",
+			name: friendlyBindingNames.unsafe,
```
Comment (RamIdeas) [inline packages/wrangler/src/config/index.ts:538]: This has been changed from "Unsafe Metadata" to "Unsafe" – was that intentional?

**Comment by CarmenPopoviciu** [comment]
@penalosa do we still want to land this?

**Review by CarmenPopoviciu — APPROVED** [review]

**Pull request merged**